### ISSUE TITLE
String interpolation in SwiftUI Text doesn't work with system locale correctly

### ISSUE THREAD

**Issue by yo1995** [Esri/arcgis-maps-sdk-swift-samples#169]
```swift
Text("Z-value: \(model.zValue.formatted(.measurement(width: .narrow).locale(Locale(identifier: "en-CA"))))")
```
Shows as "Z-value: 70m"

```swift
Text("Z-value: \(model.zValue, format: .measurement(width: .narrow).locale(Locale(identifier: "en-CA")))")
```
Shows as "Z-value: 230'"

**Comment by philium** [comment]
I suspect it is because SwiftUI views get the locale from the environment. Does this show the expected value?

```swift
Text("Z-value: \(model.zValue, format: .measurement(width: .narrow))")
    .environment(\.locale, Locale(identifier: "en-CA"))
```

**Comment by yo1995** [comment]
> Does this show the expected value?

Yes it shows the correct unit as "Z-value: 70m". It kind of makes sense that the priority is `measurement string formatting locale` > `environment locale` > `measurement value string interpolation format style locale`, but still hard to remember.

![Simulator Screenshot - iPhone 14 Pro Max - 2023-04-13 at 10 51 05](https://user-images.githubusercontent.com/9660181/231844413-7bcd087a-840b-4d63-81a8-e8d641439571.png)

I set the locale in Simulator as this - locale ID is still "en-US" while the measurement system is changed to Metric.
Using the older formatter API, the dimensions are correctly displayed as metric units, while the SwiftUI formatting style, or `formatted()` are displayed as imperial.

I'll keep digging to see if SwiftUI formatting style APIs support displaying a different measurement system than the locale's default.

**Comment by philium** [comment]
On iOS 16 and newer there is [`Locale.measurementSystem`](https://developer.apple.com/documentation/foundation/locale/4020222-measurementsystem), which sounds like what you want. It can be specified by creating a locale using `Locale.init(components:)`:

```swift
var components = Locale.Components(locale: .current)
components.measurementSystem = .metric
let locale = Locale(components: components)
```

**Comment by yo1995** [comment]
Thanks Phil. This definitely looks like the correct way to go. Though I tested in iOS 16, it doesn't behave as expected. I'll add this issue to [iOS 16 and beyond](https://github.com/Esri/arcgis-maps-sdk-swift-samples/projects/1#column-19135139) column and take a look in the future.
